consul/datadog_checks/consul/metrics.py

```diff
@@ -0,0 +1,45 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+METRIC_MAP = {
+    'consul_memberlist_degraded': 'memberlist.degraded',
+    'consul_memberlist_gossip': 'memberlist.gossip',
+    'consul_memberlist_health_score': 'memberlist.health.score',
+    'consul_memberlist_msg_alive': 'memberlist.msg.alive',
+    'consul_memberlist_msg_dead': 'memberlist.msg.dead',
+    'consul_memberlist_msg_suspect': 'memberlist.msg.suspect',
+    'consul_memberlist_probeNode': 'memberlist.probenode',
+    'consul_memberlist_pushPullNode': 'memberlist.pushpullnode',
+    'consul_memberlist_tcp_accept': 'memberlist.tcp.accept',
+    'consul_memberlist_tcp_connect': 'memberlist.tcp.connect',
+    'consul_memberlist_tcp_sent': 'memberlist.tcp.sent',
+    'consul_memberlist_udp_received': 'memberlist.udp.received',
+    'consul_memberlist_udp_sent': 'memberlist.udp.sent',
+    'consul_raft_state_leader': 'raft.state.leader',
+    'consul_raft_state_candidate': 'raft.state.candidate',
+    'consul_raft_state_apply': 'raft.apply',
+    'consul_raft_commitTime': 'raft.commitTime',
+    'consul_raft_leader_dispatchLog': 'raft.leader.dispatchLog',
+    'consul_raft_leader_lastContact': 'raft.leader.lastContact',
+    'consul_serf_events': 'serf.events',
+    'consul_serf_coordinate_adjustment_ms': 'serf.coordinate.adjustment_ms',
+    'consul_serf_member_flap': 'serf.member.flap',
+    'consul_serf_member_join': 'serf.member.join',
+    'consul_serf_member_update': 'serf.member.update',
+    'consul_serf_member_left': 'serf.member.left',
+    'consul_serf_member_failed': 'serf.member.failed',
+    'consul_serf_msgs_received': 'serf.msgs.received',
+    'consul_serf_msgs_sent': 'serf.msgs.sent',
+    'consul_serf_queue_Event': 'serf.queue.event',
+    'consul_serf_queue_Intent': 'serf.queue.intent',
+    'consul_serf_queue_Query': 'serf.queue.query',
+    'consul_serf_snapshot_appendline': 'serf.snapshot.appendLine',
+    'consul_serf_snapshot_compact': 'serf.snapshot.compact',
+    # TODO: waiting for https://github.com/hashicorp/consul/pull/8271
+    # 'consul_http': 'http',
+    # TODO: waiting for https://github.com/hashicorp/raft/pull/409
+    # 'consul_raft_replication_installSnapshot': 'raft.replication.installSnapshot',
+    # 'consul_raft_replication_heartbeat': 'raft.replication.heartbeat',
+    # 'consul_raft_replication_appendEntries_rpc': 'raft.replication.appendEntries.rpc',
+    # 'consul_raft_replication_appendEntries_logs': 'raft.replication.appendEntries.logs',
```

Comment:
Waiting for the consul PRs to be merged and having a new release. Note that we can still merge this PR and add the `http` and `raft.replication` metrics prometheus support later.

Reply:
Let's merge this PR to support the prometheus endpoint, even if some metrics are missing. We can add them later.